### PR TITLE
split process function into process_lut & process_fusion

### PR DESCRIPTION
**Pull request by rabauke**
I have split the process function into two to increase maintainability. Several issues as indicated by clang-tidy have been fixed.

### CODE REVIEW

**Comment by rabauke** [comment]
I revised my PR to fix the following UI bug: Exposure step and exposure bias sliders are not always hidden when exposure fusion is turned off, e.g., when a new module instance is created. This bug is fixed in the function `gui_init`.

File: src/iop/basecurve.c [hunk 1]
```diff
@@ -60,9 +60,9 @@ typedef struct dt_iop_basecurve_params_t
   dt_iop_basecurve_node_t basecurve[3][MAXNODES];
   int basecurve_nodes[3];
   int basecurve_type[3];
-  int exposure_fusion;    // number of exposure fusion steps
```
Comment (TurboGit) [inline src/iop/basecurve.c]: maybe this should be out of this patch-set as only reformatting.

File: src/iop/basecurve.c [hunk 2]
```diff
@@ -178,65 +180,61 @@ typedef struct basecurve_preset_t
   int filter;
 } basecurve_preset_t;
 
-#define m MONOTONE_HERMITE
-
 static const basecurve_preset_t basecurve_camera_presets[] = {
   // copy paste your measured basecurve line at the top here, like so (note the exif data and the last 1):
   // clang-format off
-
```
Comment (TurboGit) [inline src/iop/basecurve.c]: likewise, all this looks like reformatting.

File: src/iop/basecurve.c [hunk 3]
```diff
@@ -178,65 +180,61 @@ typedef struct basecurve_preset_t
...
+    // contributed by Pascal Obry
+    { "Nikon D800",           "NIKON",             "D800",                 0, FLT_MAX, {{{{ 0.000000, 0.000000 }, { 0.001773, 0.001936 }, { 0.009671, 0.009693 }, { 0.016754, 0.020617 }, { 0.024884, 0.037309 }, { 0.048174, 0.107768 }, { 0.056932, 0.139532 }, { 0.085504, 0.233303 }, { 0.130378, 0.349747 }, { 0.155476, 0.405445 }, { 0.175245, 0.445918 }, { 0.217657, 0.516873 }, { 0.308475, 0.668608 }, { 0.375381, 0.754058 }, { 0.459858, 0.839909 }, { 0.509567, 0.881543 }, { 0.654394, 0.960877 }, { 0.783380, 0.999161 }, { 0.859310, 1.000000 }, { 1.000000, 1.000000 }}}, { 20 }, { MONOTONE_HERMITE }}, 0, 1 },
   // clang-format on
 };
 static const int basecurve_camera_presets_cnt = sizeof(basecurve_camera_presets) / sizeof(basecurve_preset_t);
 
 static const basecurve_preset_t basecurve_presets[] = {
   // clang-format off
-  // smoother cubic spline curve
```
Comment (TurboGit) [inline src/iop/basecurve.c]: likewise, just reformatting. Again I'm not against it, but it makes the actual patch more difficult to read.

File: src/iop/basecurve.c [hunk 4]
```diff
@@ -105,10 +105,12 @@ int legacy_params(dt_iop_module_t *self, const void *const old_params, const int
     // start with a fresh copy of default parameters
     // unfortunately default_params aren't inited at this stage.
     *n = (dt_iop_basecurve_params_t){ {
-                                        { { 0.0, 0.0 }, { 1.0, 1.0 } },
```
Comment (TurboGit) [inline src/iop/basecurve.c]: likewise.

File: src/iop/basecurve.c [hunk 5]
```diff
@@ -105,10 +105,12 @@ int legacy_params(dt_iop_module_t *self, const void *const old_params, const int
...
+                                          { { 0.0, 0.0 }, { 1.0, 1.0 } },
                                       },
                                       { 2, 3, 3 },
-                                      { MONOTONE_HERMITE, MONOTONE_HERMITE, MONOTONE_HERMITE } , 0, 1};
```
Comment (TurboGit) [inline src/iop/basecurve.c]: likewise.

File: src/iop/basecurve.c [hunk 6]
```diff
@@ -178,65 +180,61 @@ typedef struct basecurve_preset_t
   int filter;
 } basecurve_preset_t;
 
-#define m MONOTONE_HERMITE
```
Comment (TurboGit) [inline src/iop/basecurve.c]: just a code clean-up?

File: src/iop/basecurve.c [hunk 7]
```diff
@@ -317,13 +316,13 @@ static void set_presets(dt_iop_module_so_t *self, const basecurve_preset_t *pres
       tmp.exposure_bias = 1.0f;
     }
     // add the preset.
-    dt_gui_presets_add_generic(_(presets[k].name), self->op, self->version(),
```
Comment (TurboGit) [inline src/iop/basecurve.c]: likewise. I'll stop read from here. Please remove all the reformatting out of the actual patch doing the real work.

thanks.

Comment (rabauke) [reply src/iop/basecurve.c]: Reformatting is a result of clang-format, which has reformatted the code according to the darktable formatting style. I was required to integrate clang-format into git by the darktable developers in the past. I have *not* applied these reformatting by hand.

Comment (TurboGit) [reply src/iop/basecurve.c]: That's not what I was told and what I heard. We have a reformatting script that we do run from time to time. Having all those reformatting into a PR is just noise and makes it hard to read. As-is I won't approve it anyway as I cannot see exactly what has been really changed. Too much work, sorry.

Comment (TurboGit) [reply src/iop/basecurve.c]: And to be clear, I'm not against the reformatting as long as it is on a separate commit to ease the review.

**Comment by rabauke** [comment]
I have tried to remove pure formatting changes as much as possible.

**Comment by TurboGit** [comment]
Thanks for the code clean-up.

**Pull request merged**